app/gui/view/graph-editor/src/builtin/visualization/native/text_visualization.rs

```diff
@@ -0,0 +1,469 @@
+
+                    let parent = Some(dom_entry_root.clone_ref());
+                    grid_view_entry::Params { parent }
+                })
+            );
+
+            item_width_update <- all(init, fonts_loaded, font_name, font_size);
+            item_width <- item_width_update.map(f!([]((_, _, font_name, font_size)) {
+                let font_size = *font_size;
+                let char_width = measure_character_width(font_name, font_size);
+                CHARS_PER_CHUNK as f32 * char_width
+            })).on_change();
+            item_update <- all(init, item_width, font_size);
+            text_grid.set_entries_size <+ item_update.map(f!([]((_, item_width, item_height)) {
+                Vector2::new(*item_width, *item_height)
+            }));
+
+        }
+
+        init.emit(());
+    }
+
+    /// Set the text provider.
+    pub fn set_text_provider(&self, text_provider: T) {
+        self.model.set_text_provider(text_provider);
+        self.init_frp(self.model.text_provider.borrow().as_ref().unwrap());
+    }
+}
+
+impl<T> From<TextGrid<T>> for visualization::Instance {
+    fn from(t: TextGrid<T>) -> Self {
+        Self::new(&t, &t.frp, &t.network, Some(t.model.clipping_div.clone_ref()))
+    }
+}
+
+impl<T> display::Object for TextGrid<T> {
+    fn display_object(&self) -> &display::object::Instance {
+        &self.root
+    }
+}
+
+
+
+// =================================
+// === Text Processing Utilities ===
+// =================================
+
+// Return the width of a character in the default monospaced font defined in `FONT_NAME`.
+fn measure_character_width(font_name: &str, font_size: f32) -> f32 {
+    // We expect the font to be monospaced, so we can measure the width of any character.
+    let sample_text = "█";
+
+    let canvas = web::document.create_canvas_or_panic();
+    let context = canvas.get_context("2d").unwrap().unwrap();
+    let context: CanvasRenderingContext2d = context.dyn_into().unwrap();
+    let font = format!("{font_size}px {font_name}");
+    context.set_font(&font);
+    let text_metrics = context.measure_text(sample_text).unwrap();
+    let text_length = sample_text.chars().count() as f32;
+    let width = text_metrics.width();
+    let result = width as f32 / text_length as f32;
+    result
+}
+
+fn on_fonts_loaded_event(network: &frp::Network) -> frp::Source {
+    frp::extend! { network
+            on_font_loaded <- source::<()>();
+    }
+
+    let closure = web::Closure::new(f_!(on_font_loaded.emit(())));
+    match web::document.fonts().ready() {
+        Ok(promise) => {
+            promise.then(&closure);
+        }
+        Err(e) => {
+            warn!("Could not set up font loading event because of error: {:?}.", e);
+        }
+    }
+    // This extends the lifetime of the closure which is what we want here. Otherwise, the closure
+    // would be destroyed and the callback cannot be called.
+    #[allow(clippy::forget_non_drop)]
+    mem::forget(closure);
```

Comment:
Don't use `mem:forget` if you do not need to. You can store the closure in `network` instead (it has a function for storing such things). Even better would be to store it in the visualization struct itself, so it can be properly documented. Using `mem::forget` should be the absolutely last resort.